### PR TITLE
add flowtype eslint

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers 
this adds a flowtype helper to eslint so we can start to cleanup our flow stuff. we can use this to help figure out which files don't have flow and once that list is empty we can turn it on so all files are flow typed

### CODE REVIEW

File: shared/actions/login/index.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ function waitingForResponse (waiting: boolean) : TypedAction<'login:waitingForRe
   }
 }
 
-export function navBasedOnLoginState () :AsyncAction {
+export function navBasedOnLoginState (): AsyncAction {
```
Comment (chrisnojima) [inline shared/actions/login/index.js]: now we can catch this type of stuff :)

**Comment by chrisnojima** [comment]
- [ ] need to revendor

**Comment by chrisnojima** [comment]
needs https://github.com/keybase/js-vendor-desktop/pull/19 merged

File: shared/actions/login/index.js [hunk 2]
```diff
@@ -420,8 +420,6 @@ function makeKex2IncomingMap (dispatch, getState, onBack: SimpleCB, onProvisione
     dispatch(routeAppend({parseRoute: {componentAtTop: {element}}}))
   }
 
-  let username = null
```
Comment (chrisnojima) [inline shared/actions/login/index.js]: also this is marked as not being used which seems correct to me

**Comment by chromakode** [comment]
:+1:

**Pull request merged**